pkg/envoy/lds/ingress.go

```diff
@@ -9,66 +9,59 @@ import (
 	"google.golang.org/protobuf/types/known/anypb"
 	"google.golang.org/protobuf/types/known/wrapperspb"
 
-	configv1alpha2 "github.com/openservicemesh/osm/pkg/apis/config/v1alpha2"
-
 	"github.com/openservicemesh/osm/pkg/constants"
 	"github.com/openservicemesh/osm/pkg/envoy"
 	"github.com/openservicemesh/osm/pkg/envoy/rds/route"
 	"github.com/openservicemesh/osm/pkg/errcode"
-	"github.com/openservicemesh/osm/pkg/service"
 	"github.com/openservicemesh/osm/pkg/trafficpolicy"
 )
 
-func (lb *listenerBuilder) getIngressFilterChains(svc service.MeshService) []*xds_listener.FilterChain {
-	ingressPolicy, err := lb.meshCatalog.GetIngressTrafficPolicy(svc)
-	if err != nil {
-		log.Error().Err(err).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrIngressFilterChain)).
-			Msgf("Error getting ingress filter chain for proxy with identity %s and service %s", lb.serviceIdentity, svc)
-		return nil
-	}
-
-	if ingressPolicy == nil {
-		log.Trace().Msgf("No ingress policy confiugred for proxy with identity %s and service %s", lb.serviceIdentity, svc)
+func (lb *listenerBuilder) buildIngressFilterChains() []*xds_listener.FilterChain {
+	if lb.ingressTrafficPolicies == nil {
 		return nil
```

Comment:
this looks like it gets appended to the filter chains, that's ok?

Would it be better to wrap the call around `buildIngressFilterChains` within the nil check?

Reply:
No, `filterChains = append(filterChains, lb.buildIngressFilterChains()...)` does not append nil values.